src/components/KTruncate/KTruncate.vue

```diff
@@ -0,0 +1,281 @@
+<template>
+  <div
+    ref="wrapper"
+    class="k-truncate"
+    :class="[expanded ? 'expanded' : '', `k-truncate-${isTextContent ? 'text' : 'content'}`]"
+  >
+    <div
+      ref="container"
+      class="container"
```

Comment:
suggestion: `container` is a pretty generic class that could have potential style conflicts given a host app's other styles. 

Can you rename this class something more specific, or at least prefix it like `k-truncate-container`?